contracts/Pricer.sol

```diff
@@ -260,36 +261,28 @@ contract Pricer is OpsManaged, PricerInterface {
         bytes3 _currency,
         uint256 _intendedPricePoint)
         public
-        returns (bool /* success */)
+        returns (uint256 /* total paid */)
     {
-        require(_beneficiary != address(0));
-        require(_transferAmount != 0);
-
-        if (_commissionAmount > 0) {
-            require(_commissionBeneficiary != address(0));
-        }
+        require(isValidBeneficiaryData(_beneficiary, _transferAmount,
+            _commissionBeneficiary, _commissionAmount));
 
         uint256 tokenAmount = _transferAmount;
         uint256 commissionTokenAmount = _commissionAmount;
         uint256 pricePoint = _intendedPricePoint;
-        if (_currency != 0) {
-            pricePoint = getPricePoint(_currency);
-            require(pricePoint > 0);
-            require(isPricePointInRange(_intendedPricePoint, pricePoint, pricerAcceptedMargins[_currency]));            
-            (tokenAmount, commissionTokenAmount) = getBTAmountFromCurrencyValue(pricePoint, 
-                _transferAmount, _commissionAmount);
-        }
-        
-        require(EIP20Interface(pricerBrandedToken).transferFrom(msg.sender, _beneficiary, tokenAmount));
-        if (_commissionBeneficiary != address(0)) {
-            require(EIP20Interface(pricerBrandedToken).transferFrom(msg.sender, 
-                _commissionBeneficiary, commissionTokenAmount));
+
+        // check Margin And Calculate BTAmount
+        if (_currency != "") {
+            (pricePoint, tokenAmount, commissionTokenAmount) = validateMarginAndCalculateBTAmount(_currency,
+                _intendedPricePoint, _transferAmount, _commissionAmount);
         }
-        
+
+        require(performTransfer(msg.sender, _beneficiary, tokenAmount,
+            _commissionBeneficiary, commissionTokenAmount));
+
         //Trigger Event for PaymentComplete
         Payment(_beneficiary, _transferAmount, _commissionBeneficiary, 
-            _commissionAmount, _currency, _intendedPricePoint, pricePoint);      
-        return true;
+            _commissionAmount, _currency, _intendedPricePoint, pricePoint);
+        return (tokenAmount + commissionTokenAmount);
     }
 
     /// @dev    Takes _currency; 
```

Comment:
`getPricePoint` needs to be a `public view` function; this produces a warning because Price Oracle can emit an event; but the conclusion is then to remove the event emission from `getPrice` in Price Oracle;  this should also be be specified in the Price Oracle interface

again, the interaction layer can determine that there was no valid price point based on the return value, and an event on the get function is not needed